vendor/vendor-android/ddmlib/src/main/kotlin/com/malinskiy/marathon/android/ddmlib/shell/LogcatParserListener.kt

```diff
@@ -1,21 +1,48 @@
 package com.malinskiy.marathon.android.ddmlib.shell
 
 import com.android.ddmlib.IDevice
+import com.android.ddmlib.logcat.LogCatHeader
 import com.android.ddmlib.logcat.LogCatLongEpochMessageParser
 import com.android.ddmlib.logcat.LogCatMessage
 import org.apache.commons.io.input.TailerListenerAdapter
+import java.lang.StringBuilder
 
+/**
+ * Listens for logcat lines from `adb logcat -v long -v epoch` and emits parsed LogCatMessage on receiver with message bodies grouped by the header
+ */
 class LogcatParserListener(
     private val device: IDevice,
     private val receiver: (List<LogCatMessage>) -> Unit
 ) : TailerListenerAdapter() {
 
     private val parser = LogCatLongEpochMessageParser()
+    private val messageBuffer = StringBuilder()
+    private var lastHeader: LogCatHeader? = null
 
     override fun handle(line: String?) {
```

Comment:
updated logic of the parser so now it merges all the lines with the same header (previously crash stacktrace was reported in separate logcat messages but with the same header)